### PR TITLE
fix(xray): clean stale routing references when a balancer or outbound is deleted

### PR DESCRIPTION
**Pull request by nima1024m**
## Problem

Deleting a balancer or outbound on the **Xray Settings** page left routing rules pointing at the removed entity. A dangling `balancerTag` is fatal — xray-core's `Router.Init` returns `"balancer <tag> not found"`, so the whole core fails to (re)start and **every inbound drops**. A dangling `outboundTag` black-holes matched traffic at the dispatcher. `CheckXrayConfig` only validates JSON shape, so the broken config saved cleanly and only detonated on the next core restart.

## Fix

When an outbound or balancer is deleted, the rules/balancers that reference it are repaired in the same edit, and the delete confirmation lists exactly what will change (modified vs removed):

- A rule is **kept** (dead reference dropped) if it still has a destination (`outboundTag` or `balancerTag`); it is **removed** if none remains, since a destination-less rule black-holes its matched traffic.
- Deleting an outbound **cascades**: removing it from a balancer's `selector` can empty the selector, which removes that balancer too — and then its rules are repaired the same way. `fallbackTag` and other outbounds' `dialerProxy` references are cleared, and observatories are re-synced.

Pure logic lives in `frontend/src/pages/xray/reference-cleanup.ts` (`plan*` for the preview, `apply*` for the mutation, sharing one `classifyRule` so the preview can never disagree with the mutation), wired into the balancer/outbound delete modals via a shared `DeletionImpactList`.

## Scope

- ✅ **Balancer + outbound** deletion (this PR, frontend — they live only in the client-side xray config blob, staged until Save).
- ✅ **Inbound** deletion is already handled on the backend (`RemoveInboundTagReferences`).
- ⏳ **Client `user[]`** pruning is deferred to a follow-up: a dangling `user` entry is inert in xray-core, and client records are deleted from several scattered backend paths with no single chokepoint.

## Testing

- 17 unit tests: keep/drop logic, the outbound→balancer cascade, `fallbackTag`/`dialerProxy` cleanup, disabled rules, plan/apply parity, and the tagless/edge cases.
- Gate green: typecheck, lint, 628 frontend tests, Go build, frontend build. Verified manually in the panel across all scenarios.
- A multi-angle review (correctness / cleanup / altitude / conventions) found no correctness bugs in valid configs; its top suggestions — share the rule classifier, guard tagless balancers, drop an orphaned helper — are applied.
- i18n for the impact modal added across all 13 locales.
- Manual testing

## Screenshots

{62E5F3BD-DFFA-4A28-BA22-7F2CD387BB47}
{1D7CA87F-CD11-4520-88FD-70F58EFDC8E9}
{54924F75-A7B3-483A-B3E9-D9D270E86711}

### CODE REVIEW

File: frontend/src/pages/xray/reference-cleanup.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+ * disagree. Returns null when the rule is untouched; otherwise `keeps` names the
+ * surviving destination, or is '' when none remains and the rule must be dropped.
+ */
+function classifyRule(
+  rule: RuleObject,
+  removedOutbounds: Set<string>,
+  removedBalancers: Set<string>,
+): { losesOut: boolean; losesBal: boolean; keeps: string } | null {
+  const out = typeof rule.outboundTag === 'string' ? rule.outboundTag : '';
+  const bal = typeof rule.balancerTag === 'string' ? rule.balancerTag : '';
```
[the remark this answers is not in the document]
Comment (nima1024m) [reply frontend/src/pages/xray/reference-cleanup.ts]: Thanks for the thorough review!

Good catch on the null-guard — fixed in `c30f7b4d`. `classifyRule` and the balancer loops dereferenced `rule`/`balancer` directly, while the sibling `propagateOutboundTagRename` uses optional chaining for exactly the reason you noted: `fetchXrayConfig` falls back to the unvalidated parsed object when Zod validation fails, so a stray `null` in `rules`/`balancers` can survive into the editor. I've matched that defensive style — `classifyRule` and `balancersEmptiedBy` read through optional chaining, the cleanup loop skips nullish balancer entries (which also keeps the downstream `syncObservatories` pass safe), and the `dialerProxy` walk guards the outbound. A delete on a hand-edited config with `null` entries now degrades gracefully instead of throwing, covered by a new test.

On the maintainer notes: confirmed `internal/web/dist/` is gitignored (CI rebuilds it), and the `fallbackTag = ''` vs `delete dialerProxy` asymmetry is intentional — `''` is the established no-fallback convention the balancer form and rename helper already use.
